### PR TITLE
Automate infra views ui changes

### LINKED ISSUE #3538: Automate infra views UI changes
Minor UI changes as in #1544 
```
Note in both cases the relevant information should not be viewable in the UI
if you lack permissions.
For example, a normal user could see the list of Chef Servers,
but the "Add Chef Server" (rename to "Connect Chef Server") button would not be visible.
```
#### Changes required in UI

- [ ] Rename `Add Chef Server` to `Connect Chef Server`
- [x] Chef Server detail page input padding.
- [x] Rename "org" with "organizations" for cookbooks list, orgs detail page.
- [x] Rename "Add Chef Org" with "Add Chef Organization" modal and rename button to.
- [x] Rename "Org Name" with "Name" on orgs list page.

- [x] Make Admin Key field optional on Org detail page until we will remove it.

- [x] Padding issue on Add Org modal between form fields.

- [x] Loading spinner on chef servers list and orgs list page

- [ ] Org "admin_key" is being removed from org responses need changes in UI models as well.

- [x] Wrap buttons with the `app-authorized` component show it get hidden if user lacks permission. [Here's an example](https://github.com/chef/automate/blob/master/components/automate-ui/src/app/modules/team/team-management/team-management.component.html#L32-L34)

Screenshots:

server-detail

create-org

org-detail

### PR DESCRIPTION
**Pull request by samshinde**
### :nut_and_bolt: Description: What code changed, and why?

Minor UI changes 

- [ ] ~~Rename Add Chef Server to Connect Chef Server~~ 
_--Note: Reverted  "Connect Chef Server" to "Add Chef Server"._

- [x] Chef Server detail page input padding.

- [x] Rename "org" with "organizations" for cookbooks list, orgs detail page.

- [x] Rename "Add Chef Org" with "Add Chef Organization" modal and rename button to.

- [x] Rename "Org Name" with "Name" on orgs list page.

- [x] Make Admin Key field optional on Org detail page until we will remove it.

- [x] Padding issue on Add Org modal between form fields.

- [x] Loading spinner on chef servers list and orgs list page

- [ ] ~~Org "admin_key" is being removed from org responses need changes in UI models as well.~~ 
_--Note: Skipping this for now due to some dependent code got hampered. Tracked in #3450_

- [x] Wrap buttons with the app-authorized component show it get hidden if user lacks permission.

- NOTE: Do we need to change the 'Add Chef Server' to 'Connect Chef Server' modal pop-up tittle as well?

### :chains: Related Resources 
fixes #3538 

### :+1: Definition of Done

- Added padding changes in Add Org modal between form fields.

- Renamed "org" with "organizations".

- Wrapped buttons with the app-authorized.

### :athletic_shoe: How to Build and Test the Change
STEP 1
```
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab
```
### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
addChefServer
orgModal
uichange

### CODE REVIEW

**Comment by btm** [comment]
@samshinde is  "Org "admin_key" is being removed from org responses need changes in UI models as well." unchecked because that is being done in another PR?

**Comment by btm** [comment]
@susanev I think we want the server and organization modals and buttons to all say "Connect Chef Server" and "Connect Chef Organization" yes? Because "add" sounds like "create."

Connect Chef Server:
- [X] button on /infrastructure/chef-servers (done in this PR)
- [ ] title on "add" modal
- [ ] button on "add" modal

Connect Chef Organization:
- [ ] button on /infrastructure/chef-servers/b784894d-238c-4035-ae4f-b4903cbd6566
- [ ] title on "add" modal
- [ ] button on "add" modal

**Comment by susanev** [comment]
so the ux standard is `create` when you are creating an automate object and `add` when you are adding an object that was created outside of automate. i could bring up changing all of our adds to `connect` with the ux team if you feel strongly about it. let me know.

**Comment by btm** [comment]
@susanev i think connect is better but not strongly enough. but where did "Rename Add Chef Server to Connect Chef Server" come from on #3240, because now it is Connect Chef Server on that one button, as seen in the screenshot above behind the modal, but Add in the modal.

**Comment by susanev** [comment]
i have no idea. @jonong1972 did that come from you?

**Comment by btm** [comment]
@jonong1972 @susanev looking at the edit history on #1544, it looks like I left words indicating we should do that over two months ago. I still don't remember that edit even though GitHub says I made it, so we want to change the 'Connect' button here back to 'Add' and forget I ever said anything.

**Comment by samshinde** [comment]
> @samshinde is "Org "admin_key" is being removed from org responses need changes in UI models as well." unchecked because that is being done in another PR?

@btm Yes, will do this changes with reset admin key issue https://github.com/chef/automate/issues/3450
also, I have updated the description for the same and revert the Connect chef-server label.

**Comment by btm** [comment]
@samshinde I just merged #3589 and then looked at this issue. I see it has merge conflicts and that you force-pushed recently, so these might be new merge conflicts to resolve. Sorry. :(

**Comment by samshinde** [comment]
> @samshinde I just merged #3589 and then looked at this issue. I see it has merge conflicts and that you force-pushed recently, so these might be new merge conflicts to resolve. Sorry. :(

@btm it's okay, I have resolved the merge conflicts and pushed it again Thanks! :)

File: components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.html [hunk 1]
```diff
@@ -76,40 +76,46 @@
...
         </form>
       </section>
       <section class="page-body" *ngIf="tabValue === 'orgs'">
-        <ng-container>
+        <chef-loading-spinner *ngIf="orgsListLoading" size="50"></chef-loading-spinner>
+        <ng-container *ngIf="!orgsListLoading">
           <chef-toolbar>
-            <chef-button id="create-button" primary (click)="openCreateModal('create')">Add Org</chef-button>
+            <app-authorized [allOf]="['/api/v0/infra/servers', 'post']">
+              <chef-button id="create-button" primary (click)="openCreateModal('create')">Add Chef Organization
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.html:95]: Not introduced by you I realize, but please delete this unused parameter here and in the definition.

Also delete `modalType` on chef-server-details.component.ts:L52

```suggestion
              <chef-button id="create-button" primary (click)="openCreateModal()">Add Chef Organization
```

File: components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.html [hunk 2]
```diff
@@ -76,40 +76,46 @@
...
           </div>
         </form>
       </section>
       <section class="page-body" *ngIf="tabValue === 'orgs'">
-        <ng-container>
+        <chef-loading-spinner *ngIf="orgsListLoading" size="50"></chef-loading-spinner>
+        <ng-container *ngIf="!orgsListLoading">
           <chef-toolbar>
-            <chef-button id="create-button" primary (click)="openCreateModal('create')">Add Org</chef-button>
+            <app-authorized [allOf]="['/api/v0/infra/servers', 'post']">
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.html:94]: Adding an organization needs the `infra:infraServers:update` action, which means you need to use a parameterized introspection call like this:
```suggestion
            <app-authorized [allOf]="['/api/v0/infra/servers/{server_id}/orgs', 'post', server?.id]">
```

File: components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.html [hunk 3]
```diff
@@ -76,40 +76,46 @@
...
                 <chef-td>
-                  <a [routerLink]="['/infrastructure','chef-servers', server?.id, 'org', org.id]">{{ org.name }}</a>
+                  <a [routerLink]="['/infrastructure','chef-servers', server?.id, 'organizations', org.id]">{{ org.name }}</a>
                 </chef-td>
                 <chef-td>{{ org.admin_user }}</chef-td>
                 <chef-td class="three-dot-column">
-                  <mat-select panelClass="chef-control-menu" id="menu-{{org.id}}">
-                    <mat-option data-cy="delete" (onSelectionChange)="startOrgDelete($event, org)">Delete Org</mat-option>
-                  </mat-select>
+                  <app-authorized [allOf]="['/api/v0/infra/servers', 'post']">
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.html:114]: This technically needs the same change I mentioned above. BUT: we do not yet have an efficient way to do this for every entry in a table. So suggest leave it as a reminder but comment it out, like we have done here:
https://github.com/chef/automate/blob/72325ab808f09179be86c193efd2b817d19735a4/components/automate-ui/src/app/modules/policy/list/policy-list.component.html#L29-L33

So needs to be updated as shown:
```
<!-- <app-authorized [allOf]="['/api/v0/infra/servers/{server_id}/orgs/{id}', 'delete', [server?.id, org.id]]">
```

File: components/automate-ui/src/app/modules/infra-proxy/chef-servers-list/chef-servers-list.component.html [hunk 4]
```diff
@@ -29,7 +29,9 @@
       </app-message-modal>
       <div class="page-body">
         <chef-toolbar>
-          <chef-button primary (click)="openCreateModal()">Add Chef Server</chef-button>
+          <app-authorized [allOf]="['/api/v0/infra/servers', 'post']">
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/chef-servers-list/chef-servers-list.component.html:32]: This one is good. 👍

File: components/automate-ui/src/app/modules/infra-proxy/chef-servers-list/chef-servers-list.component.html [hunk 5]
```diff
@@ -50,10 +52,12 @@
               <chef-td>{{ server.ip_address }}</chef-td>
               <chef-td>{{ server.orgs_count }}</chef-td>
               <chef-td class="three-dot-column">
-                <mat-select panelClass="chef-control-menu" id="menu-{{server.id}}">
-                  <mat-option (onSelectionChange)="startServerDelete($event, server)" data-cy="remove-server">Remove
-                    Chef Server</mat-option>
-                </mat-select>
+                <app-authorized [allOf]="['/api/v0/infra/servers', 'post']">
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/chef-servers-list/chef-servers-list.component.html:55]: Please update and comment this one out, too.
```suggestion
              <!--  <app-authorized [allOf]="['/api/v0/infra/servers/{id}', 'delete', server.id]">
```

**Comment by samshinde** [comment]
@msorens thanks for the review, I have created a PR for the same: https://github.com/chef/automate/pull/3658

**Pull request merged**